test/fixtures/cookbooks/nssm_test/recipes/install_service.rb

```diff
@@ -1,19 +1,29 @@
 include_recipe 'java_se'
+include_recipe "nssm::default"
 
-path = "#{Chef::Config[:file_cache_path]}\\selenium-server-standalone-2.53.0.jar"
+jar_path = ::File.join(Chef::Config[:file_cache_path], "selenium-server-standalone-2.53.0.jar")
 
-remote_file path do
+remote_file jar_path do
   source 'https://selenium-release.storage.googleapis.com/2.53/selenium-server-standalone-2.53.0.jar'
 end
 
 nssm 'service name' do
   program "#{node['java_se']['win_javalink']}\\java.exe"
-  args "-jar #{path}"
+  args %(-jar "#{jar_path.gsub('/', '\\')}")
   params(
-    AppDirectory: Chef::Config[:file_cache_path],
-    AppStdout: "#{Chef::Config[:file_cache_path]}\\service.log",
-    AppStderr: "#{Chef::Config[:file_cache_path]}\\error.log",
+    AppDirectory: Chef::Config[:file_cache_path].gsub('/', '\\'),
```

Comment:
What is the advantage of using ::File.join vs "#{Chef::Config[:file_cache_path]}\\service.log"?

The latter seems cleaner and I'm not sure the .gsub('/', '\\') is needed on the former since file.separator is handled by File.join.